src/main/java/column/store/api/read/Reader.java

```diff
@@ -46,6 +47,11 @@ public interface Reader extends AutoCloseable {
      */
     StringColumnReader of(StringColumn column);
 
+    /**
+     * @return a set of column names of all columns known to the reader in the form of Strings.
+     */
+    Set<String> columnNames();
```

Comment:
TBH I would prefer `Collection<Column> columns()`, as the `InMemoryReader` already supports, but I'm fine with both.

Reply:
initially i had the same idea, but in csv i currently don't intend to store the type of columns.
Alternatively i could implement `columns()` and return a collection of `StringColumns` , but didn't really like that idea, since it might give the wrong idea of the kinds of columns that are inside the csv file